storage/src/main/java/org/apache/kafka/storage/internals/log/LocalLog.java

```diff
@@ -0,0 +1,1115 @@
+                    true,
+                    dir,
+                    topicPartition,
+                    config,
+                    scheduler,
+                    logDirFailureChannel,
+                    logger);
+            if (!newSegmentBaseOffsets.contains(seg.baseOffset())) {
+                deletedNotReplaced.add(seg);
+            }
+        }
+
+        // okay we are safe now, remove the swap suffix
+        for (LogSegment segment : sortedNewSegments) {
+            segment.changeFileSuffixes(SWAP_FILE_SUFFIX, "");
+        }
+        Utils.flushDir(dir.toPath());
+
+        return deletedNotReplaced;
+    }
+
+    /**
+     * Split a segment into one or more segments such that there is no offset overflow in any of them. The
+     * resulting segments will contain the exact same messages that are present in the input segment. On successful
+     * completion of this method, the input segment will be deleted and will be replaced by the resulting new segments.
+     * See replaceSegments for recovery logic, in case the broker dies in the middle of this operation.
+     * <p>
+     * Note that this method assumes we have already determined that the segment passed in contains records that cause
+     * offset overflow.
+     * <p>
+     * The split logic overloads the use of .clean files that LogCleaner typically uses to make the process of replacing
+     * the input segment with multiple new segments atomic and recoverable in the event of a crash. See replaceSegments
+     * and completeSwapOperations for the implementation to make this operation recoverable on crashes.</p>
+     *
+     * @param segment              Segment to split
+     * @param existingSegments     The existing segments of the log
+     * @param dir                  The directory in which the log will reside
+     * @param topicPartition       The topic
+     * @param config               The log configuration settings
+     * @param scheduler            The thread pool scheduler used for background actions
+     * @param logDirFailureChannel The LogDirFailureChannel to asynchronously handle log dir failure
+     * @param logger               The logger to log messages
+     * @return List of new segments that replace the input segment
+     */
+    public static SplitSegmentResult splitOverflowedSegment(LogSegment segment,
+                                                            LogSegments existingSegments,
+                                                            File dir,
+                                                            TopicPartition topicPartition,
+                                                            LogConfig config,
+                                                            Scheduler scheduler,
+                                                            LogDirFailureChannel logDirFailureChannel,
+                                                            Logger logger) throws IOException {
```

Comment:
Yes, it was intentional. Earlier, `logPrefix` was passed so that the respective method callers could check the respective log statements in `LocalLog` logger logs by filtering with the passed log-prefixes.
So, I thought it is better to print these logs with the respective loggers instead of sending their prefixes.

If the caller method uses a different logger then it would have been hard for users to shuffle between their logs and `LocalLog` class logs. With the new change, they will get all the related logs in their respective logger logs.